datadog/resource_datadog_dashboard.go

```diff
@@ -305,13 +305,13 @@ func buildTerraformTemplateVariables(datadogTemplateVariables *[]datadogV1.Dashb
 	terraformTemplateVariables := make([]map[string]string, len(*datadogTemplateVariables))
 	for i, templateVariable := range *datadogTemplateVariables {
 		terraformTemplateVariable := map[string]string{}
-		if v, ok := templateVariable.GetNameOk(); ok {
+		if v, ok := templateVariable.GetNameOk(); ok && v != nil {
```

Comment:
Isn't prefix the only one where that case could happen?

Reply:
In theory, yes, I haven't tried torturing the API a little bit more to see if I could make the other cases also trigger. I don't mind having the check just for the prefix, but I preferred having it more consistent.